src/state/nftMarket/reducer.ts

```diff
@@ -0,0 +1,35 @@
+import { createSlice, Dispatch, PayloadAction } from '@reduxjs/toolkit'
+import { getCollections } from './helpers'
+import { State, Collection } from './types'
+
+const initialState: State = {
+  data: {
+    collections: [],
+    users: [],
+  },
+}
+
+export const NftMarket = createSlice({
+  name: 'NftMarket',
+  initialState,
+  reducers: {
+    addCollections: (state, action: PayloadAction<Collection[]>) => {
+      state.data.collections = action.payload
+    },
+  },
+})
+
+// Actions
+const { addCollections } = NftMarket.actions
+
+// Thunks
+export const fetchCollections = () => async (dispatch: Dispatch) => {
+  try {
+    const collections = await getCollections()
+    dispatch(addCollections(collections))
+  } catch (error) {
+    console.error(error)
+  }
+}
+
```

Comment:
Please use `createAsyncThunk`

Reply:
why ?

Reply:
1. Better typed
2. Comes with a `pending` and `rejected` states